### ISSUE TITLE
Fix failures and error in tests caused by recent Avocado changes

### ISSUE THREAD

**Issue by naegelyd** [chop-dbhi/serrano#270]
Running the test suite against the current HEAD on Avocado will generate the error and failures listed below. The failures are caused by changes in https://github.com/chop-dbhi/avocado/commit/0a09b59d23e817f80d3ae3a001abfd9a2588c0e2 and the error caused by changes in https://github.com/chop-dbhi/avocado/commit/d4293cc9dc4da95fb3e28b7ce823863fead886ce.

To recreate, install Avocado from HEAD on `master` and run the test_suite. You should see the following.

```
$> python test_suite.py
Creating test database for alias 'default'...
..............................F..FFF.............................E..................................................................................
======================================================================
ERROR: test_no_processor (tests.cases.resources.tests.preview.PreviewResourceProcessorTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Don/sites/serrano/tests/cases/resources/tests/preview.py", line 11, in test_no_processor
    HTTP_ACCEPT='application/json')
  File "/Users/Don/.virtualenvs/serrano/lib/python2.7/site-packages/django/test/client.py", line 473, in get
    response = super(Client, self).get(path, data=data, **extra)
  File "/Users/Don/.virtualenvs/serrano/lib/python2.7/site-packages/django/test/client.py", line 280, in get
    return self.request(**r)
  File "/Users/Don/.virtualenvs/serrano/lib/python2.7/site-packages/django/test/client.py", line 444, in request
    six.reraise(*exc_info)
  File "/Users/Don/.virtualenvs/serrano/lib/python2.7/site-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/Users/Don/.virtualenvs/serrano/lib/python2.7/site-packages/restlib2/resources.py", line 281, in __call__
    return self.dispatch(request, *args, **kwargs)
  File "/Users/Don/.virtualenvs/serrano/lib/python2.7/site-packages/restlib2/resources.py", line 292, in dispatch
    response = method_handler(request, *args, **kwargs)
  File "/Users/Don/sites/serrano/serrano/resources/preview.py", line 121, in get
    pk = output[pk_name]
TypeError: 'int' object has no attribute '__getitem__'

======================================================================
FAIL: test_export_all_pages (tests.cases.resources.tests.exporter.ExporterResourceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Don/sites/serrano/tests/cases/resources/tests/exporter.py", line 39, in test_export_all_pages
    self.assertEqual(response.status_code, codes.no_content)
AssertionError: 200 != 204

======================================================================
FAIL: test_export_equal_page_range (tests.cases.resources.tests.exporter.ExporterResourceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Don/sites/serrano/tests/cases/resources/tests/exporter.py", line 60, in test_export_equal_page_range
    self.assertEqual(response.status_code, codes.no_content)
AssertionError: 200 != 204

======================================================================
FAIL: test_export_one_page (tests.cases.resources.tests.exporter.ExporterResourceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Don/sites/serrano/tests/cases/resources/tests/exporter.py", line 46, in test_export_one_page
    self.assertEqual(response.status_code, codes.no_content)
AssertionError: 200 != 204

======================================================================
FAIL: test_export_page_range (tests.cases.resources.tests.exporter.ExporterResourceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/Don/sites/serrano/tests/cases/resources/tests/exporter.py", line 53, in test_export_page_range
    self.assertEqual(response.status_code, codes.no_content)
AssertionError: 200 != 204

----------------------------------------------------------------------
Ran 148 tests in 91.292s

FAILED (failures=4, errors=1)
Destroying test database for alias 'default'...
```

**Comment by apendleton** [comment]
I think the error is also caused by chop-dbhi/avocado@0a09b59 . Or at least, reverting it has fixed that problem for me.

**Comment by naegelyd** [comment]
Closed in #277.
